components/button/button-toggle.js

```diff
@@ -0,0 +1,83 @@
+import { css, html, LitElement } from 'lit';
+
+/**
+ * A button container component for button toggles.
+ */
+class ButtonToggle extends LitElement {
+
+	static get properties() {
+		return {
+			/**
+			 * Pressed state
+			 * @type {boolean}
+			 */
+			pressed: { type: Boolean, reflect: true }
+		};
+	}
+
+	static get styles() {
+		return css`
+			:host {
+				display: inline-block;
+			}
+			:host([hidden]) {
+				display: none;
+			}
+			::slotted(:not(d2l-button-icon, d2l-button-subtle)),
+			:host slot[name="pressed"],
+			:host([pressed]) slot[name="not-pressed"] {
+				display: none;
+			}
+			:host slot[name="not-pressed"],
+			:host([pressed]) slot[name="pressed"] {
+				display: contents;
+			}
+		`;
+	}
+
+	firstUpdated(changedProperties) {
+		super.firstUpdated(changedProperties);
+		if (this._focusOnFirstRender) {
+			this._focusOnFirstRender = false;
+			this.focus();
+		}
+	}
+
+	render() {
+		return html`
+			<slot @click="${this._handleNotPressedClick}" name="not-pressed"></slot>
+			<slot @click="${this._handlePressedClick}" name="pressed"></slot>
+		`;
+	}
+
+	focus() {
+		if (!this.hasUpdated) {
+			this._focusOnFirstRender = true;
+			return;
+		}
+
+		const elem = this.shadowRoot.querySelector(this.pressed ? 'slot[name="pressed"]' : 'slot[name="not-pressed"]').assignedNodes()[0];
+		if (!elem) {
+			throw new Error('d2l-button-toggle: no button to focus');
+		}
+
+		elem.focus();
+	}
+
+	async _handleClick(pressed) {
```

Comment:
Yeah, I was just mulling that over. I think you're right, and we should probably expose a specific event to avoid any confusion with them wiring up different handlers to each button.